sdk/ai/azure-ai-inference/src/main/java/com/azure/ai/inference/ChatCompletionsClientBuilder.java

```diff
@@ -349,19 +350,34 @@ private HttpPipeline createHttpPipeline() {
      *
      * @return an instance of ChatCompletionsAsyncClient.
      */
-    @Generated
     public ChatCompletionsAsyncClient buildAsyncClient() {
-        return new ChatCompletionsAsyncClient(buildInnerClient());
+        return new ChatCompletionsAsyncClient(buildInnerClient(), createInferenceTracer());
     }
 
     /**
      * Builds an instance of ChatCompletionsClient class.
      *
      * @return an instance of ChatCompletionsClient.
      */
-    @Generated
     public ChatCompletionsClient buildClient() {
-        return new ChatCompletionsClient(buildInnerClient());
+        return new ChatCompletionsClient(buildInnerClient(), createInferenceTracer());
+    }
+
+    private com.azure.core.util.tracing.Tracer createTracer() {
+        final String clientName = PROPERTIES.getOrDefault(SDK_NAME, "UnknownName");
+        final String clientVersion = PROPERTIES.getOrDefault(SDK_VERSION, "UnknownVersion");
+        final com.azure.core.util.LibraryTelemetryOptions telemetryOptions
+            = new com.azure.core.util.LibraryTelemetryOptions(clientName).setLibraryVersion(clientVersion)
+                .setResourceProviderNamespace("Microsoft.CognitiveServices")
+                .setSchemaUrl(com.azure.ai.inference.ChatCompletionClientTracer.OTEL_SCHEMA_URL);
+        final com.azure.core.util.TracingOptions tracingOptions
+            = this.clientOptions == null ? null : this.clientOptions.getTracingOptions();
+        return com.azure.core.util.tracing.TracerProvider.getDefaultProvider()
+            .createTracer(telemetryOptions, tracingOptions);
+    }
```

Comment:
The "Verify TypeSpec Code Generation" CI verification stage (the check for non-generated partial updates) is currently strict. So, we need to keep the builder changes like this (to avoid imports/class fields etc..) until partial update checks improves.